crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -404,6 +415,36 @@ where
 
         Ok(())
     }
+
+    pub fn shared_data(&self) -> SharedData {
+        SharedData {
+            sync_observer: self.sync_notifier.subscribe(),
+        }
+    }
+}
+
+/// Shared data for the gas price service.
+#[derive(Debug, Clone)]
+pub struct SharedData {
+    /// Allows to observe the sync state.
+    sync_observer: SyncStateObserver,
+}
+
+impl SharedData {
+    /// Waits until the gas price service has synced
+    /// with current l2 block height
+    pub async fn await_synced(&self) -> anyhow::Result<()> {
```

Comment:
let's add another function too, `await_synced_until` which accepts a block height and checks internally until the block height is reached